MapboxDirections/MBRouteStep.swift

```diff
@@ -382,7 +436,11 @@ public class RouteStep: NSObject, NSSecureCoding {
     
     internal init(finalHeading: CLLocationDirection?, maneuverType: ManeuverType?, maneuverDirection: ManeuverDirection?, maneuverLocation: CLLocationCoordinate2D, name: String?, coordinates: [CLLocationCoordinate2D]?, json: JSONDictionary) {
         transportType = TransportType(description: json["mode"] as! String)
-        destinations = json["destinations"] as? String
+        
+        let road = Road(name: name, ref: json["ref"] as? String, destination: json["destinations"] as? String)
+        names = road.names
+        codes = road.codes ?? road.destinationCodes
+        destinations = road.destinations
```

Comment:
I assume these are properties on `self.`? Either these should be set that way, or below assignments should not have `self.=` either (like `self.intersections =` -> `intersections =`)